### DISCUSSION TITLE
Bosch Compress 7000i with HPC410 and CR10

### DISCUSSION THREAD

**Discussion by rici44** [emsesp/EMS-ESP32#2119] [Q&A]
Hi community. I have Bosch Compress 7000i managed with HPC410 which is included in inside unit and CR10 on the wall. My EMS-ESP32 gw is showing all devices on Dashboard but in Home Assistant CR10 is missing. Why is it like this? Is it able to control CR10 from Home Assistant? 

HPC410 in inside unit
![HPC 410 detail 01](https://github.com/user-attachments/assets/b1fe3292-fe97-4e17-958a-72acd4387a14)

Dashboard
<img width="981" alt="Screenshot 2024-10-22 at 08 16 19" src="https://github.com/user-attachments/assets/00462cd2-1c66-4d7c-8a54-2ab3a0ae97b3">

Home Assistant
<img width="1142" alt="Screenshot 2024-10-22 at 08 17 49" src="https://github.com/user-attachments/assets/d1e8ae06-2f7f-4e3f-82fc-67acee3a80b1">

**Comment by proddy** [comment]
Are you sure its an CR10? Please attach your support info (https://emsesp.org/Support/) and if you can upgrade to the latest 3.7.0 dev version as we're no longer supporting 3.6.5.

**Reply by rici44** [reply]
I am not sure, so my question. It could also be CR10H because there is also humidity. But why is it showing RC100H and only in Dashboard and not in HA? I am outside of the house so could not upgrade now. Please find attached support info 
[support info.txt](https://github.com/user-attachments/files/17472356/support.info.txt)

**Reply by proddy** [reply]
You're on version 3.6.4 which is a year old. I think you'll find everything should work after you've upgraded to 3.7.0. Make sure you make backups of the settings as the firmware partitions have changed. Also big changes in naming (HA too) so you have a bit of work to do.

**Reply by rici44** [reply]
Is it secure to make the update over webgui to ../system/upload? 
Which version should I choose? I got **ESP32 S3**. Is version **EMS-ESP-3_7_0-dev_46-ESP32_S3.bin** correct?

<img width="272" alt="Screenshot 2024-10-22 at 11 29 13" src="https://github.com/user-attachments/assets/fe660816-1d4e-4650-ace8-e691122031e8">

**Reply by proddy** [reply]
yes, take the S3 bin and either use the Flash tool with USB or you can try the Web OTA, although it's not secure. There isn't enough bandwidth on the ESP32 to support HTTP/S.

**Reply by rici44** [reply]
I did update over OTA to Firmware Version 3.7.0-dev.46 but devices are the same as previous 

Dashbo
<img width="1132" alt="Screenshot 2024-10-22 at 14 33 54" src="https://github.com/user-attachments/assets/a9c9816b-9d2a-4c99-8a36-bf75733760ec">
ard

HA
<img width="1143" alt="Screenshot 2024-10-22 at 14 34 09" src="https://github.com/user-attachments/assets/987b9ee2-e381-43f5-861c-de45a75a763b">

**Reply by proddy** [reply]
could you send a new version of the support info, so we can see which telegrams are being sent/received from the RC10

**Reply by rici44** [reply]
please find the attached file 
[support info.json](https://github.com/user-attachments/files/17476417/support.info.json)

**Reply by rici44** [reply]
but no devices available 
<img width="1108" alt="Screenshot 2024-10-22 at 15 00 46" src="https://github.com/user-attachments/assets/4f748a6c-38b6-489f-a3eb-81e15670c23d">

**Reply by proddy** [reply]
Ok, now we're up to date. What do you want to control via the CR10? As far as I know, it's just a remote interface to the HPC410 thermostat. Is there something specific you are after?

**Reply by rici44** [reply]
My intention was to have clear info about all connected devices in Dashboard and inside Home Assistant, because in dashboard there are 3 (boiler/HP, thermostat, thermostat) and in HA there are only 2 (boiler, thermostat). I needed this to understand what is possible to do and with which device it has to be done. For example how to manage boiler, over HPC410 or directly in Boiler device? Maybe is my lack of understanding what is causing this.

**Reply by proddy** [reply]
It can be complicated. I don't know what half the captured entities mean. EMS-ESP is just the middle-man translating what we find on the EMS bus to human-readable data and functions. The rule I normally use is "if you don't know what it is, you probably don't need it". Try and think of a usage first, and then go looking how. The easiest way to control your boiler is via the HPC410.

Why the CR10 is not showing in HA is weird. I think the MQTT data is messed up as the names are also incorrect. I would use MQTTExplorer and delete everything ems-esp related under homeassistant, and reboot EMS-ESP

**Reply by MichaelDvP** [reply]
@rici44 There is only one master thermostat that controls the heating. The CR10H (same as RC100H) is only measuring the room temp/humi and send this to the master (except sensor calibration which is stored in the CR10). Also it is a remote and every change you do on the CR10 is just send as command to the master. 
We show the CR10 in web for reference, but the values are also in the master thermostat. For mqtt we send only a single topic `thermostat_data` containing **all** entities. So HA gets only the entities from that.

If you use a thermostat you **can't** control the boiler direct, it will be overwritten by the thermostat. You have to control the thermostat (e.g. seltemp), the thermostat calculates the required flowtemp and send it to the boiler (or mixer if you have one, then mixer calculates the needed boiler headertemp and sends to the boiler).
For the boiler you can change the parameters, e.g. hysteresis, limits, etc, but not the heating values that the thermostat sends.
